### PR TITLE
MEED-431: add total supply public endpoint 

### PR DESCRIPTION
**Pull request by ArfaouiMohamedAmine**
This change will introduce a new REST endpoint to retrieve Total Supply Value of Meeds Token. This endpoint can be used and consulted from other third party exchange sites.

### CODE REVIEW

File: deeds-dapp-service/src/main/java/io/meeds/deeds/web/rest/MeedTokenMetricController.java [hunk 1]
```diff
@@ -57,4 +57,12 @@ public ResponseEntity<BigDecimal> getTotalLockedValue() {
             .body(totalValueLocked);
   }
 
+  @GetMapping("/total")
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/web/rest/MeedTokenMetricController.java]:
```suggestion
  @GetMapping("/supply")
```

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java [hunk 2]
```diff
@@ -81,6 +82,18 @@ public BigDecimal getCirculatingSupply() {
...
+  /**
+   * Retrieves total Meeds supply, reading the value of the ERC20.totalSupply() Meeds contract
+   *
+   * @return {@link BigDecimal} for most recent computed total supply
+   *           value
+   */
+  public BigDecimal getTotalSupply() {
+    MeedTokenMetric lastMetric = getLastMetric();
+    BigDecimal totalSupply = lastMetric.getTotalSupply();
+    return totalSupply.setScale(8, RoundingMode.CEILING);
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: why is this needed ?

Comment (ArfaouiMohamedAmine) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: because we need 8 decimals format

Comment (boubaker) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: it wasn't mentioned explicitly here neither for other endpoints, so please keep it simple by including all decimals

**Review by boubaker — APPROVED** [review]
